File: 品味Java的21種設計模式/dp_cpp/chainofresponsibility/example6/SaleMgr.cpp
#include "SaleMgr.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace chainofresponsibility
			{
				namespace example6
				{

					bool SaleMgr::sale(std::string user, std::string customer, SaleModel *saleModel)
					{
						//进行真正的业务逻辑处理
						puts(user+"保存了"+customer+"购买 "+saleModel+" 的销售数据");
						return true;
					}
				}
			}
		}
	}
}

File: 品味Java的21種設計模式/dp_java/chainofresponsibility/example6/SaleMgr.java
package cn.javass.dp.chainofresponsibility.example6;
/**
 * 真正处理销售的业务功能的职责对象
 */
public class SaleMgr extends SaleHandler {
    public boolean sale(String user, String customer, SaleModel saleModel) {
        //进行真正的业务逻辑处理
        System.out.println(user + "保存了" + customer + "购买 " + saleModel + " 的销售数据");
        return true;
    }
}
